geonode/upload/api/tests.py

```diff
@@ -233,35 +235,65 @@ def test_rest_uploads(self):
-        resp, data = rest_upload_by_path(fname, self.client)
-        self.assertEqual(resp.status_code, 201)
-
-        url = reverse("uploads-list")
-        # Anonymous
-        self.client.logout()
-        response = self.client.get(url, format="json")
-        self.assertEqual(response.status_code, 200)
-        self.assertEqual(len(response.data), 5)
-        self.assertEqual(response.data["total"], 0)
-        # Pagination
-        self.assertEqual(len(response.data["uploads"]), 0)
-        logger.debug(response.data)
+        layer_name = "relief_san_andres"
+        try:
+            self._cleanup_layer(layer_name=layer_name)
+            # Try to upload a good raster file and check the session IDs
+            fname = os.path.join(GOOD_DATA, "raster", "relief_san_andres.tif")
+            resp, data = rest_upload_by_path(fname, self.client)
+            self.assertEqual(resp.status_code, 201)
+
+            url = reverse("uploads-list")
+            # Anonymous
+            self.client.logout()
+            response = self.client.get(url, format="json")
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(len(response.data), 5)
+            self.assertEqual(response.data["total"], 0)
+            # Pagination
+            self.assertEqual(len(response.data["uploads"]), 0)
+            logger.debug(response.data)
+        except Exception:
+            if resp.json().get("errors"):
+                layer_name = resp.json().get("errors")[0].split("for : ")[1].split(",")[0]
+        finally:
+            self._cleanup_layer(layer_name)
 
     @override_settings(CELERY_TASK_ALWAYS_EAGER=True)
     def test_rest_uploads_non_interactive(self):
         """
         Ensure we can access the Local Server Uploads list.
         """
-        # Try to upload a good raster file and check the session IDs
-        fname = os.path.join(GOOD_DATA, "raster", "relief_san_andres.tif")
-        resp, data = rest_upload_by_path(fname, self.client, non_interactive=True)
-        self.assertEqual(resp.status_code, 201)
-
-        exec_id = data.get("execution_id", None)
-        _exec = ExecutionRequest.objects.get(exec_id=exec_id)
-        self.assertEqual(_exec.status, "finished")
+        layer_name = "relief_san_andres"
+        try:
+            self._cleanup_layer(layer_name=layer_name)
+            # Try to upload a good raster file and check the session IDs
+            fname = os.path.join(GOOD_DATA, "raster", "relief_san_andres.tif")
+            resp, data = rest_upload_by_path(fname, self.client, non_interactive=True)
+            self.assertEqual(resp.status_code, 201)
+            exec_id = data.get("execution_id", None)
+            _exec = ExecutionRequest.objects.get(exec_id=exec_id)
+            self.assertEqual(_exec.status, "finished")
+        except Exception:
+            if resp.json().get("errors"):
+                layer_name = resp.json().get("errors")[0].split("for : ")[1].split(",")[0]
```

Comment:
@mattiagiupponi you're using `reps` inside an exception, so it will be undefined here.

Reply:
Si i notice, i already fix it on my local instance